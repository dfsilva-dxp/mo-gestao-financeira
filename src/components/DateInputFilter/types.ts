export type DefaultValues = {
  default: string;
  formatted: string;
};
