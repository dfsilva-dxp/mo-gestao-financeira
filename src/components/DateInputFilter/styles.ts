import styled, { css } from "styled-components";

export const Wrapper = styled.section`
  ${({ theme }) => css`
    display: flex;
    align-items: center;
    justify-content: center;
    gap: ${theme.spacings.xxsmall};

    [type="month"] {
      max-width: 1.4rem;
      background: transparent url("./img/calendar-white-icon.svg") center
        center/contain no-repeat;
      border: none;
      outline: 0;

      &:active,
      &:focus {
        background: transparent url("./img/calendar-pink-icon.svg") center
          center/contain no-repeat;
      }
    }
    [type="month"]::-webkit-inner-spin-button {
      display: none;
    }
    [type="month"]::-webkit-calendar-picker-indicator {
      opacity: 0;
      cursor: pointer;
    }
  `}
`;

export const Label = styled.label`
  ${({ theme }) => css`
    color: ${theme.colors.gray};
    font-size: ${theme.font.sizes.small};
    text-transform: capitalize;
    font-weight: ${theme.font.weight.bold};
  `}
`;
