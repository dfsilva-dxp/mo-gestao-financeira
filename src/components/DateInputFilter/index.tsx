import { ChangeEvent, useEffect, useReducer } from "react";

import { useDate } from "hooks";

import { reducer } from "utils";

import { Label, Wrapper } from "./styles";
import { DefaultValues } from "./types";

const INITIAL_VALUES: DefaultValues = {
  default: "2020-01",
  formatted: "jan. de 2020"
};

export default function DateInputFilter() {
  const [date, setDate] = useReducer(reducer, INITIAL_VALUES);

  const { format } = useDate();

  useEffect(() => {
    const d = new Date();

    const formatted = format({
      date: d,
      locale: "pt-br",
      pattern: "MMM. [de] YYYY"
    });

    const defaultValue = format({
      date: d,
      locale: "pt-br",
      pattern: "YYYY-MM"
    });

    setDate({
      default: defaultValue,
      formatted
    });
  }, [format]);

  const handleChange = (e: ChangeEvent<HTMLInputElement>) => {
    const { value } = e.target;

    const formatted = format({
      date: value,
      locale: "pt-br",
      pattern: "MMM. [de] YYYY"
    });

    setDate({
      default: value,
      formatted
    });
  };

  return (
    <Wrapper>
      <Label htmlFor="date-filter">{date.formatted}</Label>
      <input
        type="month"
        name="date-filter"
        id="date-filter"
        aria-label="data-filter"
        value={date.default}
        onChange={handleChange}
      />
    </Wrapper>
  );
}
