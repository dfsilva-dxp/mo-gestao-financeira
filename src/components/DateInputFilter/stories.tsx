import { Story, Meta } from "@storybook/react/types-6-0";
import { ComponentProps } from "react";

import DateInputFilter from ".";

export default {
  title: "Components/DateInputFilter",
  component: DateInputFilter,
  argTypes: {},
  parameters: {
    layout: "centered"
  }
} as Meta;

export const Basic: Story<ComponentProps<typeof DateInputFilter>> = () => (
  <DateInputFilter />
);
