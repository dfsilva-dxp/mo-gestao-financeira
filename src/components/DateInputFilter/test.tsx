import { screen, fireEvent } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { renderWithTheme } from "utils/tests/helpers";

import DateInputFilter from ".";

describe("<DateInputFilter />", () => {
  it("should render with label", () => {
    const { container } = renderWithTheme(<DateInputFilter />);

    expect(screen.getByLabelText(/jun\. de 2022/i)).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });

  it("should is accessible by tab", () => {
    renderWithTheme(<DateInputFilter />);

    const input = screen.getByLabelText(/jun\. de 2022/i);

    expect(document.body).toHaveFocus();

    userEvent.tab();
    expect(input).toHaveFocus();
  });

  it("should change display when change input value", () => {
    renderWithTheme(<DateInputFilter />);

    const input = screen.getByLabelText(/data-filter/i);

    expect(input).toBeInTheDocument();

    fireEvent.change(input, { target: { value: "2022-07" } });

    expect(screen.getByLabelText(/jul\. de 2022/i)).toBeInTheDocument();
  });
});
