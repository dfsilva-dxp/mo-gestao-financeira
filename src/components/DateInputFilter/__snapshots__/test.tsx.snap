// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DateInputFilter /> should render with label 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  gap: 0.8rem;
}

.c0 [type="month"] {
  max-width: 1.4rem;
  background: transparent url("./img/calendar-white-icon.svg") center center/contain no-repeat;
  border: none;
  outline: 0;
}

.c0 [type="month"]:active,
.c0 [type="month"]:focus {
  background: transparent url("./img/calendar-pink-icon.svg") center center/contain no-repeat;
}

.c0 [type="month"]::-webkit-inner-spin-button {
  display: none;
}

.c0 [type="month"]::-webkit-calendar-picker-indicator {
  opacity: 0;
  cursor: pointer;
}

.c1 {
  color: #797D9A;
  font-size: 1.4rem;
  text-transform: capitalize;
  font-weight: 700;
}

<section
  class="c0"
>
  <label
    class="c1"
    for="date-filter"
  >
    jun. de 2022
  </label>
  <input
    aria-label="data-filter"
    id="date-filter"
    name="date-filter"
    type="month"
    value="2022-06"
  />
</section>
`;
